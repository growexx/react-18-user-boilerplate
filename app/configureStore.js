import { configureStore } from '@reduxjs/toolkit';
import { createBrowserHistory } from 'history';
import { createReduxHistoryContext } from 'redux-first-history';

import { repoApi } from './containers/HomePage/reposApiSlice';
import { appSlice } from './containers/App/slice';
import { apiSlice } from './apiSlice';
import languageSlice from './containers/LanguageProvider/slice';
import loginSlice from './containers/Auth/Login/slice';
import { authApi } from './containers/Auth/Login/authApiSlice';
import twoFactorAuthSlice from './containers/Auth/TwoFactorAuthentication/slice';

const { createReduxHistory, routerMiddleware, routerReducer } =
  createReduxHistoryContext({ history: createBrowserHistory() });

const store = configureStore({
  reducer: {
    router: routerReducer,
    app: appSlice,
    language: languageSlice,
    login: loginSlice,
    twoFactorAuth: twoFactorAuthSlice,
    repos: repoApi.reducer,
    auth: authApi.reducer,
    [apiSlice.reducerPath]: apiSlice.reducer,
  },
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware(),
    apiSlice.middleware,
    routerMiddleware,
  ],
  devTools: process.env.NODE_ENV !== 'production',
});

const history = createReduxHistory(store);

export { store, history };
