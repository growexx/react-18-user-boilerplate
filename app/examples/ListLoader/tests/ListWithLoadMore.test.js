import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { HistoryRouter as Router } from 'redux-first-history/rr6';
import { act } from 'react-dom/test-utils';
import history from 'utils/history';
import request from 'utils/request';
import configureStore from '../../../configureStore';
import ListWithLoadMore from '../ListWithLoadMore';
let globalStore;
jest.mock('utils/request');
const componentWrapper = () =>
  render(
    <Provider store={globalStore}>
      <IntlProvider locale="en">
        <Router history={history}>
          <ListWithLoadMore />
        </Router>
      </IntlProvider>
    </Provider>,
  );

describe('<ListWithLoadMore />', () => {
  beforeAll(() => {
    const { store } = configureStore({});
    globalStore = store;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render and match the snapshot with call LoadMore', async () => {
    request.mockImplementation(() =>
      Promise.resolve({
        status: 1,
        results: [
          { id: '1', name: { last: 'test name 1' }, email: 'test@example.com' },
          { id: '2', name: { last: 'test name 2' }, email: 'test@example.co' },
          { id: '3', name: { last: 'test name 3' }, email: 'test@example.cm' },
          { id: '4', name: { last: 'test name 4' }, email: 'test@example.om' },
          {
            id: '5',
            name: { last: 'test name 5' },
            email: 'test@example.co.in',
          },
        ],
      }),
    );

    const { container, debug } = componentWrapper();
    expect(container.firstChild).toMatchSnapshot();

    await waitFor(() => {
      debug(container.querySelector('button'));
      expect(request).toHaveBeenCalledTimes(1);
    });
    act(() => {
      fireEvent.click(container.querySelector('button'));
    });
    expect(request).toHaveBeenCalled();
  });
});
