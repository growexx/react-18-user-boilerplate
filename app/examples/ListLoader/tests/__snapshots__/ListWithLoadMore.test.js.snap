// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListWithLoadMore /> should render and match the snapshot with call LoadMore 1`] = `
<div
  class="ant-list ant-list-split demo-loadmore-list"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <ul
        class="ant-list-items"
      >
        <li
          class="ant-list-item"
        >
          <div
            class="ant-skeleton ant-skeleton-with-avatar ant-skeleton-active"
          >
            <div
              class="ant-skeleton-header"
            >
              <span
                class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
              />
            </div>
            <div
              class="ant-skeleton-content"
            >
              <ul
                class="ant-skeleton-paragraph"
              >
                <li />
                <li
                  style="width: 61%;"
                />
              </ul>
            </div>
          </div>
        </li>
        <li
          class="ant-list-item"
        >
          <div
            class="ant-skeleton ant-skeleton-with-avatar ant-skeleton-active"
          >
            <div
              class="ant-skeleton-header"
            >
              <span
                class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
              />
            </div>
            <div
              class="ant-skeleton-content"
            >
              <ul
                class="ant-skeleton-paragraph"
              >
                <li />
                <li
                  style="width: 61%;"
                />
              </ul>
            </div>
          </div>
        </li>
        <li
          class="ant-list-item"
        >
          <div
            class="ant-skeleton ant-skeleton-with-avatar ant-skeleton-active"
          >
            <div
              class="ant-skeleton-header"
            >
              <span
                class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
              />
            </div>
            <div
              class="ant-skeleton-content"
            >
              <ul
                class="ant-skeleton-paragraph"
              >
                <li />
                <li
                  style="width: 61%;"
                />
              </ul>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
